File: LMS_CMS_Frontend/src/app/Models/LMS/daily-performance.ts
import { StudentPerformance } from "./student-performance";

export class DailyPerformance {
  constructor(
    public id: number = 0,
    public subjectID: number = 0,
    public studentID: number = 0,
    public comment: string = "",
    public studentPerformance: StudentPerformance[] = [],
    public insertedByUserId: number = 0
  ) {}
}

File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/LMS/DailyPerformance.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.LMS
{
    public class DailyPerformance : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        public string? Comment { get; set; }
        [ForeignKey("Subject")]
        public long SubjectID { get; set; }
        public Subject Subject { get; set; }

        [ForeignKey("Student")]
        public long StudentID { get; set; }
        public Student Student { get; set; }
        public ICollection<StudentPerformance> StudentPerformance { get; set; } = new HashSet<StudentPerformance>();
    }
}
